doc/services/ipc/ipc_service/ipc_service.rst

```diff
@@ -0,0 +1,252 @@
+Registration of the endpoints is left to the user - usually at run time.
+
+The API does not mandate a way for the backend to create instances but it is
+strongly recommended to use the DT to retrieve the configuration parameters for
+an instance. Currently each backend define its own DT compatible that is used
+to configure interface at boot time.
+
+Usage
+*****
+
+Simple data exchange
+====================
+
+In order to send data between domains/CPUs an endpoint must be registered onto
+an instance. Note that before registering an endpoint, instance must be opened
+with :c:func:`ipc_service_open_instance` function.
+
+.. code-block:: c
+
+   #include <zephyr/include/ipc_service.h>
+
+   static void bound_cb(void *priv)
+   {
+      /* Endpint bounded */
+   }
+
+   static void recv_cb(const void *data, size_t len, void *priv)
+   {
+      /* Data received */
+   }
+
+   static struct ipc_ept_cfg ept0_cfg = {
+      .name = "ept0",
+      .cb = {
+         .bound    = bound_cb,
+         .received = recv_cb,
+      },
+   };
+
+   int main(void)
+   {
+      const struct device *inst0;
+      struct ipc_ept ept0;
+      int ret;
+
+      inst0 = DEVICE_DT_GET(DT_NODELABEL(ipc0));
+      ret = ipc_service_open_instance(inst0);
+      ret = ipc_service_register_endpoint(inst0, &ept0, &ept0_cfg);
+
+      /* Wait for endpint bound (bound_cb called) */
+
+      unsigned char message[] = "hello world";
+      ret = ipc_service_send(&ept0, &message, sizeof(message));
+   }
+
+Data exchange with no copy API
+==============================
+
+If the backend supports no copy API you can use it to directly write and read
+to and from shared memory regions.
+
+.. code-block:: c
+
+   #include <zephyr/include/ipc_service.h>
+   #include <stdint.h>
+   #include <string.h>
+
+   static struct ipc_ept ept0;
+
+   static void bound_cb(void *priv)
+   {
+      /* Endpint bounded */
+   }
+
+   static void recv_cb_nocopy(const void *data, size_t len, void *priv)
+   {
+      int ret;
+
+      ret = ipc_service_hold_rx_buffer(&ept0, (void *)data);
+      /* Process directly or put the buffer somewhere else and release. */
+      ret = ipc_service_release_rx_buffer(&ept0, (void *)data);
+   }
+
+   static struct ipc_ept_cfg ept0_cfg = {
+      .name = "ept0",
+      .cb = {
+         .bound    = bound_cb,
+         .received = recv_cb,
+      },
+   };
+
+   int main(void)
+   {
+      const struct device *inst0;
+      int ret;
+
+      inst0 = DEVICE_DT_GET(DT_NODELABEL(ipc0));
+      ret = ipc_service_open_instance(inst0);
+      ret = ipc_service_register_endpoint(inst0, &ept0, &ept0_cfg);
+
+      /* Wait for endpint bound (bound_cb called) */
+      void *data;
+      unsigned char message[] = "hello world";
+      uint32_t len = sizeof(message);
+
+      ret = ipc_service_get_tx_buffer(&ept0, &data, &len, K_FOREVER);
+
+      memcpy(data, message, len);
+
+      ret = ipc_service_send_nocopy(&ept0, data, sizeof(message));
+   }
+
+Backends
+********
```

Comment:
I will move this in a separate file (to have one documentation file per backend) and just list here the backends with the reference to the associated .rst  file
Otherwise, the file could become very large after describing all the backends and potentially adding new ones.

Reply:
sorry my comment is not well placed consider it for the` ICMsg backend `folder

Reply:
Okay. Will do